### PR TITLE
priority for minion_blackout_whitelist by grains

### PR DESCRIPTION
**Pull request by pimiento**
### What does this PR do?
For a purpose to make blackout_mode managed by minion not by master, I need that pillar config for `minion_blackout_whitelist` couldn't manage the same option on minion. So I changed `elif` to `if`, so I **also**  check grains.

### Previous Behavior
Master can add option `minion_blackout` for given minion in a **pillar** data, so the minion.py will check `minion_blackout_whitelist` from a **pillar** dict and won't check in **grains**

### New Behavior
Even if it is the `minion_blackout` options in a **pillar**, minion.py will check these options in **grains** then..

### Tests written?

No

### Commits signed with GPG?

No

### CODE REVIEW

File: salt/minion.py [hunk 1]
```diff
@@ -1594,7 +1594,8 @@ def _thread_return(cls, minion_instance, opts, data):
                     # this minion is blacked out. Only allow saltutil.refresh_pillar and the whitelist
                     if function_name != 'saltutil.refresh_pillar' and function_name not in whitelist:
                         minion_blackout_violation = True
-                elif minion_instance.opts['grains'].get('minion_blackout', False):
+                # minion_blackout_whitelist by grains has higher priority than by pillar
```
Comment (DmitryKuzmenko) [inline salt/minion.py]: I disagree with this comment. By the code now there is no any priority, if blackout mode is enabled and function isn't whitelisted it's blocked.

**Review by DmitryKuzmenko — CHANGES_REQUESTED** [review]
The change described and implemented here looks reasonable for me. Just disagree with the comment.

**Review by DmitryKuzmenko — APPROVED** [review]
I'm ok here now. Thank you @pimiento .

**Comment by pimiento** [comment]
Ok, I've changed comment

**Review by terminalmage — APPROVED** [review]

**Comment by max-arnold** [comment]
@pimiento Could you please document this feature here: https://docs.saltstack.com/en/latest/topics/blackout/ ?

You can submit the PR against 2019.2 branch.

**Pull request merged**